packages/block-editor/src/components/link-control/index.js

```diff
@@ -201,14 +201,16 @@ function LinkControl( {
 		useCreatePage( createSuggestion );
 
 	useEffect( () => {
-		if (
-			forceIsEditingLink !== undefined &&
-			forceIsEditingLink !== isEditingLink
-		) {
-			setIsEditingLink( forceIsEditingLink );
+		if ( forceIsEditingLink === undefined ) {
+			return;
 		}
-		// Todo: bug if the missing dep is introduced. Will need a fix.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
+
+		setIsEditingLink( ( prevIsEditing ) => {
+			if ( prevIsEditing === forceIsEditingLink ) {
+				return prevIsEditing;
+			}
```

Comment:
Can someone confidence check me here on this logic 😓

Reply:
Updated. I wonder about the `undefined` check above now as well 🤔

Reply:
I understand that `forceIsEditingLink` should be `undefined` when the components want to derive the initial `isEditingLink` state based on value.